### ISSUE TITLE
BatchMatmul on GPU Wrong Result

### ISSUE THREAD

**Issue by kor01** [tensorflow/tensorflow#33560]
<em>Please make sure that this is a bug. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:bug_template</em>

**System information**
-  Linux Ubuntu 18.04
- TensorFlow installed from (source or binary): pip install tensorflow-gpu
- TensorFlow version (use command below): 2.0
- Python version: 3.7.3
- CUDA/cuDNN version:
- GPU model and memory:Quadro P4000 8GB

**Describe the current behavior**

batchmatmul on gpu gives wrong result.

**Describe the expected behavior**

batchmatmul should give correct result.

**Code to reproduce the issue**
```python
import numpy as np
import tensorflow as tf
H, W = 480, 640
xs = tf.random.normal((H, W, 3))
m = tf.random.normal((3, 3))
with tf.device('/gpu:0'):
  mvgpu = tf.linalg.matvec(m[None, None, None, :, : ], xs[None])

with tf.device('cpu:0'):
  mvcpu = tf.linalg.matvec(m[None, None, None, :, : ], xs[None])

diff = mvgpu - mvcpu
print(diff) # a lot of large differences
assert not np.allclose(diff, 0)
```

**Other info / logs**
python diff.py 
tf.Tensor(
[[[[ 0.          0.          0.        ]
   [ 0.          0.          0.        ]
   [ 0.          0.          0.        ]
   ...
   [ 0.          0.          0.        ]
   [ 0.          0.          0.        ]
   [ 0.          0.          0.        ]]

  [[ 0.          0.          0.        ]
   [ 0.          0.          0.        ]
   [ 0.          0.          0.        ]
   ...
   [ 0.          0.          0.        ]
   [ 0.          0.          0.        ]
   [ 0.          0.          0.        ]]

  [[ 0.          0.          0.        ]
   [ 0.          0.          0.        ]
   [ 0.          0.          0.        ]
   ...
   [ 0.          0.          0.        ]
   [ 0.          0.          0.        ]
   [ 0.          0.          0.        ]]

  ...

  [[ 2.7437444  -2.3577256   3.3067198 ]
   [ 1.0874022  -0.7325933   2.4926019 ]
   [-2.3672104   1.6497741  -2.1795664 ]
   ...
   [-1.4023315   0.68545413 -1.9975882 ]
   [-0.44164103 -1.6766946  -0.6477224 ]
   [ 2.5612988   0.7603723   4.286979  ]]

  [[-0.9761815   1.5332515  -0.12404668]
   [ 1.871354   -0.73664904 -1.0545558 ]
   [-1.2774239   1.6571116  -2.36473   ]
   ...
   [ 2.4847538  -1.5389507   1.5169847 ]
   [ 2.2117858  -3.3593345  -0.15468699]
   [ 0.82464755 -2.0175495  -0.11506134]]

  [[ 2.8697317   0.54362947 -0.10824746]
   [ 1.6531861  -4.2324843   0.97695297]
   [ 4.3966837  -3.0250916   1.8032213 ]
   ...
   [ 4.5854487  -2.0374475  -1.1027236 ]
   [-2.7206469   0.864337   -1.3582373 ]
   [-0.98220587  0.53226703 -4.277097  ]]]], shape=(1, 480, 640, 3), dtype=float32)

**Comment by gadagashwini** [comment]
@kor01, Thanks for reporting this issue,
I tried replicating the issue on colab with TF 2.0, it throws assertion error.  Please take a look at the [gist](https://colab.sandbox.google.com/gist/gadagashwini/24dfd6669d3be4e3a40c45e561195a11/untitled217.ipynb). Thanks!

**Comment by kor01** [comment]
> @kor01, Thanks for reporting this issue,
> I tried replicating the issue on colab with TF 2.0, it throws assertion error. Since the `diff` is `0`, `assert not` throws exception. Let me know if i misunderstood your issue. Please take a look at the [gist](https://colab.sandbox.google.com/gist/gadagashwini/24dfd6669d3be4e3a40c45e561195a11/untitled217.ipynb). Thanks!

I run the script and replicate the error on my laptop (Quadro P4000 8GB) and my desktop computer( TITAN V  & Titan X Maxwell). However I could not replicate on K80 device in colab cloud. I have python 3.7 on my two computers whereas colab has python 3.6. It could be device or python version dependent. 

The problem is also replicated in:
https://github.com/tensorflow/tensorflow/issues/31166

the problem characteristics are the same with mine. A portion of computation is done correctly another portion is left with zeros.

**Comment by ymodak** [comment]
This is fixed in latest tf-nightly version '2.1.0-dev20200110' which uses cuda 10.1. (tested in google colab) Thanks!
